client/client.go

```diff
@@ -137,24 +153,30 @@ func (c *client) GetID(ctx context.Context) (*ID, error) {
 	}
 	for _, e := range batch {
 		if e.Error != nil {
+			zlog.Err(e.Error).Msg("GetID: BatchCallContext")
 			log.Printf("Method %q failed: %v\n", e.Method, e.Error)
 		}
 	}
 	netID := new(big.Int)
 	if _, ok := netID.SetString(netIDStr, 10); !ok {
-		return nil, fmt.Errorf("invalid net_version result %q", netIDStr)
+		err := fmt.Errorf("invalid net_version result %q", netIDStr)
+		zlog.Err(err).Msg("GetID: netID.SetString(netIDStr, 10)")
```

Comment:
Shouldn't be writing log messages in here, caller can log if they want. Same goes for probably all the logging you added.

Reply:
All the logging that is just prior to returning an error anyways.